components/eamxx/src/share/field/field.cpp

```diff
@@ -92,7 +121,7 @@ subfield (const std::string& sf_name, const ekat::units::Units& sf_units,
         "Error! Subview dimension index must be either 0 or 1.\n");
 
   // Create identifier for subfield
-  FieldIdentifier sf_id(sf_name,lt.clone().strip_dim(idim),sf_units,id.get_grid_name());
+  FieldIdentifier sf_id(sf_name,lt.clone().strip_dim(idim),sf_units,id.get_grid_name(),id.data_type());
```

Comment:
This change is unrelated, but we were not passing the data type when creating a subfield.

Reply:
It's ok to add it here.